segmentstore/server/src/test/java/io/pravega/segmentstore/server/containers/StreamSegmentContainerTests.java

```diff
@@ -1513,7 +1521,60 @@ public void testStartOffline() throws Exception {
 
             container.stopAsync().awaitTerminated();
         }
+    }
 
+    /**
+     * Tests the ability to register extensions.
+     */
+    @Test
+    public void testExtensions() throws Exception {
+        // Configure extension.
+        val operationProcessed = new CompletableFuture<SegmentOperation>();
+        AtomicInteger count = new AtomicInteger();
+        val writerProcessor = new TestWriterProcessor(op -> {
+            count.incrementAndGet();
+            if (!operationProcessed.isDone()) {
+                operationProcessed.complete(op);
+            }
+        });
+
+        val extension = new AtomicReference<TestSegmentContainerExtension>();
+        SegmentContainerFactory.CreateExtensions createExtensions = (container, executor) -> {
+            Assert.assertTrue("Already created", extension.compareAndSet(null,
+                    new TestSegmentContainerExtension(Collections.singleton(writerProcessor))));
+            return Collections.singletonMap(TestSegmentContainerExtension.class, extension.get());
+        };
+
+        @Cleanup
+        val context = new TestContext(createExtensions);
+
+        // Verify extension is initialized when the SegmentContainer is started.
+        context.container.startAsync().awaitRunning();
+        Assert.assertTrue("Extension not initialized.", extension.get().initialized.get());
+
+        // Verify getExtension().
+        val p = context.container.getExtension(TestSegmentContainerExtension.class);
+        Assert.assertEquals("Unexpected result from getExtension().", extension.get(), p);
+
+        // Verify Writer Segment Processors are properly wired in.
+        String segmentName = getSegmentName(0);
+        byte[] data = getAppendData(segmentName, 0);
+        context.container.createStreamSegment(segmentName, null, TIMEOUT).join();
+        context.container.append(segmentName, data, null, TIMEOUT).join();
+        val rawOp = operationProcessed.get(TIMEOUT.toMillis(), TimeUnit.MILLISECONDS);
+        Assert.assertTrue("Unexpected operation type.", rawOp instanceof CachedStreamSegmentAppendOperation);
```

Comment:
I don't actually understand how we are checking that the segment processors are properly wired in. Is this only checking that they are not creating any deviation of the correct behavior?